xdsl/transforms/reconcile_unrealized_casts.py

```diff
@@ -0,0 +1,117 @@
+import itertools
+from collections import deque
+from typing import Iterable
+from warnings import warn
+
+from xdsl.dialects import builtin
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext
+from xdsl.ir.core import Use
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class RemoveUselessConversionCast(RewritePattern):
```

Comment:
why not `ReconcileUnrealizedCasts` like in MLIR? Isn't it pretty much the same functionality?
moreover, I'm not really fond of the "useless" in the name; there was utility once but now it's a noop.